packages/docs/docs/tutorial/part3.md

```diff
@@ -0,0 +1,125 @@
+---
+sidebar_position: 3
+---
+
+# Tutorial Part 3 - Constraining the output
+
+When generating text from an LLM, it is often useful to constrain the output to a particular
+format, such as JSON or YAML. AI.JSX provides a number of components that can be used to
+achieve this.
+
+In the following example, we're going to have the LLM generate a character sheet for a
+fantasty role-playing game, and then ensure that the output is formatted as either
+JSON or YAML.
+
+## Generating a plain-text character sheet
+
+Let's start with a component called `<CharacterGenerator>` that will generate a list of
+fields
+
+```tsx filename="packages/tutorial/src/part3.tsx"
```

Comment:
For some reason, the `filename` doesn't consistently show up. I can't tell if we're doing it wrong or Docusaurus has a bug.

<img width="341" alt="image" src="https://github.com/fixie-ai/ai-jsx/assets/829827/62026111-17ad-42c3-9f77-ce15ce3b7028">